Lib/tempfile.py

```diff
@@ -100,6 +100,8 @@ def _infer_return_type(*args):
     for arg in args:
         if arg is None:
             continue
+        elif isinstance(arg, _os.PathLike):
```

Comment:
Why this explicit test is needed? Aren't high-level function fail in any case when pass path-like object as prefix or suffix?